src/Microsoft.AspNetCore.Server.IIS/Core/IISNativeApplication.cs

```diff
@@ -31,7 +32,17 @@ public void StopCallsIntoManaged()
             IntPtr requestContext,
             IntPtr shutdownContext)
         {
-            NativeMethods.HttpRegisterCallbacks(_nativeApplication, requestHandler, shutdownHandler, onAsyncCompletion, requestContext, shutdownContext);
+            NativeMethods.HttpRegisterCallbacks(_nativeApplication, requestHandler, shutdownHandler, onAsyncCompletion, requestContext, shutdownContext, out var fLoggingEnabled);
+
+            if (!fLoggingEnabled)
```

Comment:
could the assumption fLoggingEnabled == !PipePutputRedirection ever change?

Reply:
It could, but we can always change the value we return in native to true/false appropriately.